evalml/pipelines/time_series_classification_pipelines.py

```diff
@@ -61,21 +60,41 @@ def fit(self, X, y):
         y = self._encode_targets(y)
 
         X_t = self._compute_features_during_fit(X, y)
-        if X_t.empty:
-            raise RuntimeError("Pipeline computed empty features during call to .fit. This means "
-                               "that either 1) you passed in X=None to fit and don't have a DelayFeatureTransformer "
-                               "in your pipeline or 2) you do have a DelayFeatureTransformer but gap=0 and max_delay=0. "
-                               "Please add a DelayFeatureTransformer or change the values of gap and max_delay")
 
         y_shifted = y.shift(-self.gap)
         X_t, y_shifted = drop_rows_with_nans(X_t, y_shifted)
         self.estimator.fit(X_t, y_shifted)
         return self
 
+    def _estimator_predict_helper(self, features, y, return_probabilities):
+        """Helper to call estimator.predict(features, y) or estimator.predict(features) depending on the estimator.
+
+        If return_probabilities=True, then estimator.predict_proba is called instead.
+        """
+        method = self.estimator.predict_proba if return_probabilities else self.estimator.predict
+        if self.estimator.predict_uses_y:
+            return method(features, y)
+        return method(features)
+
+    def _estimator_predict(self, features, y):
+        """Get estimator predictions.
+
+        This helper passes y as an argument if needed by the estimator.
+        """
+        return self._estimator_predict_helper(features, y, return_probabilities=False)
+
+    def _estimator_predict_proba(self, features, y):
+        """Get estimator predicted probabilities.
+
+        This helper passes y as an argument if needed by the estimator.
+        """
+        return self._estimator_predict_helper(features, y, return_probabilities=True)
```

Comment:
@freddyaboulton could we do this
```python
def _estimator_predict(self, features, y):
        """Get estimator predictions.
        This helper passes y as an argument if needed by the estimator.
        """
        y_arg = None
        if self.estimator.predict_uses_y:
            y_arg = y
        return self.estimator.predict(features, y_arg, return_probabilities=False)
```
And similar for predict_proba?

Reply:
Yes we can! I originally thought we couldn't because some estimators like `XGBoost` don't have `y` as an argument in their predict signature but the metaclass adds it!